src/pcre2/_cy.pyx

```diff
@@ -202,20 +208,23 @@ def pattern_name_dict(PCRE2Code code not None):
         const uint8_t *name_table, *name
         uint32_t name_count, name_entry_size
         int idx, offset
+        object encoding
 
     # Get name table related information
     raise_from_rc(pcre2_pattern_info(code.ptr, PCRE2_INFO_NAMECOUNT, &name_count))
     raise_from_rc(pcre2_pattern_info(code.ptr, PCRE2_INFO_NAMEENTRYSIZE, &name_entry_size))
     raise_from_rc(pcre2_pattern_info(code.ptr, PCRE2_INFO_NAMETABLE, &name_table))
 
+    encoding = "UTF-8" if pattern_is_utf(code) else "Latin-1"
+
     # Convert byte table to dictionary mapping group names to numbers
     name_dict = {}
     for idx in range(name_count):
         # Name table is structured with first two bytes of name table contain group number
-        # followed by name string (gaurunteed to be ASCII for non-unicode patterns)
+        # followed by name string (which is in Latin-1 for non-unicode patterns)
```

Comment:
It would be possible for someone to have a build of PCRE2 which uses Latin-1, or indeed any single-byte encoding (for example see the way Debian bundles python-pcre2 using the system PCRE2 library). But Latin-1 is the canonical choice because the 256 bytes values are mapped to the first 256 Unicode characters.